cypress/e2e/09-multiplex_testing.cy.js

```diff
@@ -1,55 +1,65 @@
+import {
+  generateMultiplexDevice,
+  generatePatient,
+  loginHooks,
+  testNumber,
+} from "../support/e2e";
 import { graphqlURL } from "../utils/request-utils";
+import {
+  cleanUpPreviousRunSetupData,
+  cleanUpRunOktaOrgs,
+  setupMultiplexDevice,
+  setupRunData,
+  setupPatient,
+} from "../utils/setup-utils";
+
+const specRunName = "spec09";
+const currentSpecRunVersionName = `${testNumber()}-cypress-${specRunName}`;
 
 loginHooks();
 describe("Testing with multiplex devices", () => {
-  let patient, facility, multiplexDeviceName;
+  let facilityId, patientId;
+  const patient = generatePatient();
+  const multiplexDevice = generateMultiplexDevice();
 
   before(() => {
-    cy.makePOSTRequest({
-      operationName: "WhoAmI",
-      variables: {},
-      query:
-        "query WhoAmI {\n  whoami {\n organization {\n    facilities {\n      id\n      name\n      __typename\n    }\n    __typename\n  }\n} \n}",
-    }).then((res) => {
-      facility = res.body.data.whoami.organization.facilities[0];
-      cy.makePOSTRequest({
-        operationName: "GetPatientsByFacility",
-        variables: {
-          facilityId: facility.id,
-          pageNumber: 0,
-          pageSize: 1,
-          archivedStatus: "UNARCHIVED",
-        },
-        query:
-          "query GetPatientsByFacility($facilityId: ID!, $pageNumber: Int!, $pageSize: Int!, $archivedStatus: ArchivedStatus, $namePrefixMatch: String) {\n  patients(\n    facilityId: $facilityId\n    pageNumber: $pageNumber\n    pageSize: $pageSize\n    archivedStatus: $archivedStatus\n    namePrefixMatch: $namePrefixMatch\n  ) {\n    internalId\n    firstName\n    lastName\n    middleName\n    birthDate\n    isDeleted\n    role\n    lastTest {\n      dateAdded\n      __typename\n    }\n    __typename\n  }\n}",
-      }).then((res) => {
-        patient = res.body.data.patients[0];
-      });
-    });
+    cy.task("getSpecRunVersionName", specRunName).then(
+      (prevSpecRunVersionName) => {
+        if (prevSpecRunVersionName) {
+          cleanUpPreviousRunSetupData(prevSpecRunVersionName);
+          cleanUpRunOktaOrgs(prevSpecRunVersionName);
+        }
+        let data = {
+          specRunName: specRunName,
+          versionName: currentSpecRunVersionName,
+        };
+        cy.task("setSpecRunVersionName", data);
 
-    cy.task("getMultiplexDeviceName").then((name) => {
-      multiplexDeviceName = name;
-    });
+        setupRunData(currentSpecRunVersionName).then((result) => {
+          facilityId = result.body.data.addFacility.id;
+        });
+        setupPatient(currentSpecRunVersionName, patient).then((result) => {
+          patientId = result.internalId;
+        });
+        setupMultiplexDevice(currentSpecRunVersionName, multiplexDevice);
+      },
+    );
   });
 
   beforeEach(() => {
     cy.intercept("POST", graphqlURL, (req) => {
       aliasGraphqlOperations(req);
     });
+  });
 
-    // remove a test for the patient if it exists
-    cy.makePOSTRequest({
```

Comment:
I think we may still want this for test robustness. If someone gets left in a queue (maybe because of a test interruption), we'd want to remove them to get the tests to work

Reply:
Wouldn't the rerun create a completely new facility and therefore a fresh queue each time?

Reply:
referring here to something like a test retry which I don't think retrigger the before hook

<img width="1016" alt="Screenshot 2024-02-07 at 12 04 02 PM" src="https://github.com/CDCgov/prime-simplereport/assets/29645040/31ad6e1e-0101-40eb-8df9-a32624baa397">

Reply:
Ah okay I thought you meant rerun as referring to rerunning the full spec as opposed to the retry attempts within the test, yeah I can add this back in then